### ISSUE TITLE
Clean up accessibility warnings

### ISSUE THREAD

**Issue by theetrain** [carbon-design-system/carbon-components-svelte#1929]
Rarely, Svelte's a11y warnings are incorrect such as when it cannot detect valid children due to usage of `<slot>`. However, all warnings should be resolved properly.

## Outcomes

- [ ] Remove all usage of `autofocus` and guide developers to set focus manually with `ref`
- [ ] Add a `ref` prop to all applicable components (part of #1621)
- [ ] Remove and resolve `<!-- svelte-ignore -->` comments wherever possible
